### DISCUSSION TITLE
Is it a block of dead code?

### DISCUSSION THREAD

**Discussion by Yi-Zhai** [CoRE-RG/FiCo4OMNeT#15] [General]
Hello community,

I was wondering whether the line may cause dead code?
https://github.com/CoRE-RG/FiCo4OMNeT/blob/545afef2eb01ae117d303fba9f011e48f2a25569/src/fico4omnet/applications/can/sink/CanTrafficSinkAppBase.cc#L55

Perhaps should "==" instead of "=" in this line?

**Comment by Musteblume** [comment]
Hi Yi-Zhai,

it is a variable declaration, initialization, and test if it is null within the if statement. It is done very frequently in our models.
It can be translated as: "if msg is of type CanDataFrame*"

Best regards
Philipp
